vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/dataview/CheckboxGroupListDataView.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2000-2020 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.checkbox.dataview;
+
+import com.vaadin.flow.component.checkbox.CheckboxGroup;
+import com.vaadin.flow.data.provider.AbstractListDataView;
+import com.vaadin.flow.data.provider.DataController;
+
+/**
+ * {@link CheckboxGroup} component list data view implementation which takes into
+ * account the component-specific and common list data API.
+ *
+ * @param <T>
+ *        data type
+ */
+public abstract class CheckboxGroupListDataView<T>
```

Comment:
I'm not sure why do we need this ? I think checkboxgroup should just always provide a list data view from its API, so why can't we just have `CheckboxGroupDataView<T> extends DataView<T>, ListDataView<T>` ? I feel this would then just be a totally useless abstraction level

Reply:
Method `T getItemOnIndex(int index)` has been moved to `CheckboxGroupDataView` from `DataView` so as to have more appropriate name for concrete component. I would keep it as it is now, because component team may request some any other API to be added into `CheckboxGroupDataView` interface, or we may decided to add something after the demo and DX.

Reply:
Sorry I still don't get why this should be an abstract class instead of an marker interface `interface CheckboxGroupListDataView<T> extends CheckboxGroupDataView<T>, ListDataView<T>`.

What would be added to this class instead of the interface or the `...Impl` class ? Are we going to add that now or next ?